docsrc/workflow.rst

```diff
@@ -0,0 +1,193 @@
+If you are running GPU hardware and wish to use the OpenCL framework to speed up matrix operations,
+you must set the C++ compiler flag **STAN_OPENCL**.
+For high-level within-chain parallelization using the Stan language `reduce_sum` function,
+it's necessary to set the C++ compiler flag **STAN_THREADS**.  While any value can be used,
+we recommend the value ``True``.
+
+For example, given Stan program named 'proc_parallel.stan', you can take
+advantage of both kinds of parallelization by specifying the compiler options when instantiating
+the model:
+
+.. code-block:: python
+
+    proc_parallel_model = CmdStanModel(
+        stan_file='proc_parallel.stan',
+        cpp_options={"STAN_THREADS": True, "STAN_OPENCL": True},
+    )
+
+
+Assemble input and initialization data
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+CmdStan is file-based interface, therefore all model input and
+initialization data must be supplied as JSON files, as described in the
+CmdStan User's Guide:
+<https://mc-stan.org/docs/cmdstan-guide/json.html>.
+
+CmdStanPy inference methods allow inputs and initializations
+to be specified as in-memory Python dictionary objects
+which are then converted to JSON via the utility method :func:`cmdstanpy.write_stan_json`.
+This method should be used to create JSON input files whenever
+these inputs contain either a collection compatible with
+numpy arrays or pandas.Series.
+
+
+Run the CmdStan inference engine
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For each CmdStan inference method, there is a corresponding method on the CmdStanModel class.
+
+* The :meth:`cmdstanpy.CmdStanModel.sample` method runs Stan's
+  `HMC-NUTS sampler <https://mc-stan.org/docs/reference-manual/hamiltonian-monte-carlo.html>`_
+  and returns a :class:`cmdstanpy.CmdStanMCMC` object which contains
+  a sample from the posterior distribution of the model conditioned on the data.
+
+* The :meth:`cmdstanpy.CmdStanModel.variational` method runs Stan's
+  `Automatic Differentiation Variational Inference (ADVI) algorithm <https://mc-stan.org/docs/reference-manual/vi-algorithms-chapter.html>`_
+  and returns   a :class:`cmdstanpy.CmdStanVB` object which contains
+  an approximation the posterior distribution in the unconstrained variable space.
+
+* The :meth:`cmdstanpy.CmdStanModel.optimize` runs one of
+  `Stan's optimization algorithms <https://mc-stan.org/docs/reference-manual/optimization-algorithms-chapter.html>`_
+  to find a mode of the density specified by the Stan program.  It
+  returns a :class:`cmdstanpy.CmdStanMLE` object.
+
+* The :meth:`cmdstanpy.CmdStanModel.generate` method runs Stan's
+  `generate_quantities method <https://mc-stan.org/docs/cmdstan-guide/standalone-generate-quantities.html>`_
+  which generates additional quantities of interest from a mode. Its take an existing sample as input and
+  uses the parameter estimates in the sample to run the Stan program's ``generated quantities`` block.
```

Comment:
add link to https://mc-stan.org/docs/reference-manual/program-block-generated-quantities.html